include/ck_tile/ops/gemm/pipeline/tile_gemm_traits.hpp

```diff
@@ -3,23 +3,124 @@
 
 #pragma once
 
+#include "ck_tile/core.hpp"
+
 namespace ck_tile {
 
-template <bool kPadA_,
-          bool kPadB_,
-          bool kPadC_,
+template <bool kPadM_,
+          bool kPadN_,
+          bool kPadK_,
+          typename BlockGemmShape_,
+          typename ADataType_,
+          typename BDataType_,
+          typename CDataType_,
           typename ALayout_,
           typename BLayout_,
           typename CLayout_>
 struct TileGemmTraits
 {
-    static constexpr bool kPadA = kPadA_;
-    static constexpr bool kPadB = kPadB_;
-    static constexpr bool kPadC = kPadC_;
+    using BlockGemmShape = remove_cvref_t<BlockGemmShape_>;
+
+    static constexpr bool kPadM = kPadM_;
+    static constexpr bool kPadN = kPadN_;
+    static constexpr bool kPadK = kPadK_;
+
+    static constexpr int _VectorSize = 16;
+
+    static constexpr index_t kBlockSize = BlockGemmShape::NumWarps * get_warp_size();
 
     using ALayout = ALayout_;
     using BLayout = BLayout_;
     using CLayout = CLayout_;
+
+    using ADataType = remove_cvref_t<ADataType_>;
+    using BDataType = remove_cvref_t<BDataType_>;
+    using CDataType = remove_cvref_t<CDataType_>;
+
+    CK_TILE_HOST_DEVICE static constexpr auto GetAlignmentA()
```

Comment:
I think the `GetAlignmentX()` functions should move to policy.

Reply:
I changed the code and put that in the the GemmPipeline Problem so that I could be called based on the pipeline distribution.